packages/api-openfin/src/main-process.ts

```diff
@@ -0,0 +1,144 @@
+import MessageService from './message-service';
+
+// Code that is "evaled" when the main window has been opened, sets up
+// all the InterApplicationBus listeners for window events to keep track of state
+const mainWindowCode = () => {
+  const childTree = [];
+
+  const addParent = (parentName, windowName, tree) => {
+    let win = tree.find(w => w.name === parentName);
+    if (win) {
+      win.children.push({
+        name: windowName,
+        children: []
+      });
+      return true;
+    } else {
+      return tree.some((childWin) => {
+        return addParent(parentName, windowName, childWin.children);
+      });
+    }
+  };
+
+  const getParentWindow = (name, parentName, tree) => {
+    let win = tree.find(w => w.name === name);
+    if (win) {
+      return parentName;
+    } else {
+      let parentName = null;
+      tree.some((childWin) => {
+        parentName = getParentWindow(name, childWin.name, childWin.children);
+        return parentName !== null;
+      });
+      return parentName;
+    }
+  };
+
+  const getChildWindows = (name, tree) => {
+    let win = tree.find(w => w.name === name);
+    if (win) {
+      return win.children.map(c => c.name);
+    } else {
+      let childNames = [];
+      tree.some((childWin) => {
+        childNames = getChildWindows(name, childWin.children);
+        return childNames && childNames.length > 0;
+      });
+      return childNames;
+    }
+  };
+
+  const deleteWindow = (name, tree) => {
+    let index = tree.findIndex(w => w.name === name);
+    if (index >= 0) {
+      tree.splice(index, 1);
+    } else {
+      tree.forEach((childWin) => {
+        deleteWindow(name, childWin.children);
+      });
+    }
+  };
+
+  const closeChildren = (uuid) => {
+    const childUuids = getChildWindows(uuid, childTree);
+    childUuids.forEach((child) => {
+      fin.desktop.Application.wrap(child).close(true);
+      closeChildren(child);
+    });
+  }
+
+  fin.desktop.InterApplicationBus.subscribe('*', 'ssf-new-window', (data) => {
```

Comment:
Ah, just checked the docs, I hadn't realised that this filter was for the application UUID, I thought it was just some sort of topic string.

Does that mean you will have to filter these messages to ensure you only respond to those relating to your current ContainerJS application?